File: Java Barcode Scanner ME/app/src/main/java/de/dimedis/mobileentry/fragments/menus/AbsSettingsMenuFragment.java
package de.dimedis.mobileentry.fragments.menus;

import android.os.Bundle;
import android.view.View;

import androidx.annotation.Nullable;

import de.dimedis.mobileentry.SettingsController;
import de.dimedis.mobileentry.SettingsControllerImpl;
import de.dimedis.mobileentry.fragments.BaseFragment;
import de.dimedis.mobileentry.util.dynamicres.DynamicString;

public class AbsSettingsMenuFragment extends BaseFragment {
    protected SettingsController mSettings;

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        DynamicString.scan(view);
        super.onViewCreated(view, savedInstanceState);
        init();
    }

    protected void onClickOK() {
        mSettings.save();
        back();
    }

    protected void onClickCancel() {
        back();
    }

    protected void init() {
        mSettings = SettingsControllerImpl.getDefault(getActivity());
        mSettings.saveTemporary();
    }

    @Override
    public void onDetach() {
        super.onDetach();
        mSettings.loadTemporary();
    }
}

File: Kotlin Migration/NTMobileEntry/app/src/main/java/de/dimedis/mobileentry/fragments/menus/AbsSettingsMenuFragment.kt
package de.dimedis.mobileentry.fragments.menus

import android.view.*

import android.os.Bundle
import de.dimedis.mobileentry.SettingsController
import de.dimedis.mobileentry.fragments.BaseFragment
import de.dimedis.mobileentry.util.dynamicres.DynamicString

open class AbsSettingsMenuFragment : BaseFragment() {
    protected var mSettings: SettingsController? = null
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        DynamicString.scan(view)
        super.onViewCreated(view, savedInstanceState)
        init()
    }

    protected fun onClickOK() {
        mSettings!!.save()
        back()
    }

    protected fun onClickCancel() {
        back()
    }

    protected open fun init() {
        mSettings = activity?.let { SettingsController.getDefault(it) }
        mSettings!!.saveTemporary()
    }

    override fun onDetach() {
        super.onDetach()
        mSettings!!.loadTemporary()
    }
}